kedro_mlflow/config/kedro_mlflow_config.py

```diff
@@ -131,13 +151,45 @@ def setup(self, context):
 
         self._export_credentials(context)
 
+        self._register_request_header_provider(context)
         # we set the configuration now: it takes priority
         # if it has already be set in export_credentials
         mlflow.set_tracking_uri(self.server.mlflow_tracking_uri)
         mlflow.set_registry_uri(self.server.mlflow_registry_uri)
 
         self._set_experiment()
 
+    def _register_request_header_provider(self, context: KedroContext):
+        # this is a specific trick to deal with expiring tokens for authentication, see https://github.com/Galileo-Galilei/kedro-mlflow/issues/357
+        if self.server.request_header_provider.type is not None:
+            # it type is none, there is nothing to register
+            request_header_provider_class = load_obj(
+                self.server.request_header_provider.type
+            )
+
+            if not issubclass(request_header_provider_class, RequestHeaderProvider):
+                raise ValueError(
+                    f"request_header_provider.type='{self.server.request_header_provider.type}' should be a sublass of 'mlflow.tracking.request_header.abstract_request_header_provider', got {request_header_provider_class} instead."
+                )
+
+            # we allow the user to instantiate the class with parameters, including the kedro context for advanded retrieval of kedro objects
+            init_kwargs = (
+                {
+                    **self.server.request_header_provider.init_kwargs,
+                    "kedro_context": context,
+                }
+                if self.server.request_header_provider.pass_context
+                else self.server.request_header_provider.init_kwargs
+            )
+
+            request_header_provider_instance = request_header_provider_class(
+                **init_kwargs
+            )
+            # cannot use the "register" method because it expects a callable class, not the instantiated object, but we need to pass arguments
+            mtrr._request_header_provider_registry._registry.append(
```

Comment:
FYI, you can pass `lambda: YourClass(<params from closure>)` to the `_request_header_provider_registry.register` as it only expects a parameter-less callable.